### DISCUSSION TITLE
I'm having trouble understanding why `loads` on my graphql mutation isn't working (no object found for ...).

### DISCUSSION THREAD

**Discussion by DennisMartinez** [rmosolgo/graphql-ruby#4520] [Q&A] [answered]
Hey Everyone. I'm working through the mutations documentation, trying to implement the autoload arguments functionally using `loads` but the system is having trouble finding the object. I keep running into the following error:

```
"No object found for `userId: \"1\"`"
```

Here's my test mutation. It takes a user_id in, loads the user, and then returns the user.
```
module Mutations
  class AddUser < BaseMutation
    field :user, Types::UserType, null: false

    argument :user_id, ID, loads: Types::UserType

    def resolve(user:)
      { user: }
    end
  end
end
```

What's interesting, is, I can get it to work by adding this method to the `AddUser` mutation above. I don't know if that is correct or not as I'm not seeing it in the documentation examples (although mentioned).
```
def object_from_id(_type, global_id, _query_ctx)
  User.find(global_id)
end
```
Do I need to set object_from_id on every mutation? I may be misunderstanding what I need to do with this method. Anything else I should read up on in order to better understand why I can't load an object?

---

I've created a test repo with my issue https://github.com/DennisMartinez/graphql_mutation_test.

Documentation / issue reports I've read referenced in trying to understand this issue:

- https://graphql-ruby.org/schema/object_identification.html
- https://graphql-ruby.org/mutations/mutation_classes.html#auto-loading-arguments
- https://graphql-ruby.org/mutations/mutation_authorization.html#loading-and-authorizing-objects
- https://github.com/rmosolgo/graphql-ruby/issues/3869. I couldn't get the proposed solution to work, but I did add a `users` query to the schema just in case.

Additional notes
- I've updated my resolve_type to acknowledge the `UserType` https://github.com/DennisMartinez/graphql_mutation_test/blob/master/app/graphql/graphql_mutation_test_schema.rb#L18
- I've applied the relay node to my user type https://github.com/DennisMartinez/graphql_mutation_test/blob/master/app/graphql/types/user_type.rb#L5

**Comment by rmosolgo** [comment, marked as answer]
Hi! Sorry for the trouble. By default, `loads:` uses the `def self.object_from_id` method defined on your _Schema_, which is here: 

https://github.com/DennisMartinez/graphql_mutation_test/blob/86133c4adeb275a7d554a8a47bff12372ee15097/app/graphql/graphql_mutation_test_schema.rb#L38-L42

When you add `def object_from_id` to your mutation class, it overrides that method from the schema -- which is why it starts working again. 

How about updating `def self.object_from_id` in the schema definition? I think that would make it work the way you expect. Give it a try and let me know what you find!

**Reply by DennisMartinez** [reply]
Hey @rmosolgo, thanks for the clarification! I see now what the problem was. I was overriding the id from the Node, which provides the global id. I was instead using the DB id. It looks like this is mentioned in the first sentence of the documentation 😅. I appreciate the help!

https://graphql-ruby.org/schema/object_identification.html#uuid-fields

```
module Types
  class UserType < Types::BaseObject
    implements GraphQL::Types::Relay::Node

    # field :id, ID, null: false <-- Problem
    field :created_at, GraphQL::Types::ISO8601DateTime, null: false
    field :updated_at, GraphQL::Types::ISO8601DateTime, null: false
  end
end
```
